examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1405 @@
+//! addresses as operators. An operator of some address is allowed to transfer
+//! any tokens owned by this address.
+//!
+//! This contract also contains an example of a function to be called when
+//! receiving tokens. In which case the contract will forward the tokens to
+//! the contract owner.
+//! 
+//! This function is not very useful and is only there to showcase a simple
+//! implementation of a token receive hook.
+//! 
+//! The token can be configured such that it pays royalties.  The settings for
+//! whether royalties are paid and how much are configured in the State initialisation.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+use concordium_cis2::*;
+use concordium_std::*;
+
+/// The baseurl for the token metadata, gets appended with the token ID as hex
+/// encoding before emitted in the TokenMetadata event.
+const TOKEN_METADATA_BASE_URL: &str = "https://some.example/token/";
+
+/// List of supported standards by this contract address.
+const SUPPORTS_STANDARDS: [StandardIdentifier<'static>; 2] =
+    [CIS0_STANDARD_IDENTIFIER, CIS2_STANDARD_IDENTIFIER];
+
+// Types
+
+/// The parameter for the contract function `mint` which mints a number of
+/// token types and/or amounts of tokens to a given address.
+#[derive(Serial, Deserial, SchemaType)]
+struct InterfaceParams {
+    owner:  i32     // this is 4 bytes
+}
+
+/// Contract token ID type.
+/// To save bytes we use a small token ID type, but is limited to be represented
+/// by a `u8`.
+type ContractTokenId = TokenIdU8;
+
+/// Contract token amount type.
+type ContractTokenAmount = TokenAmountU64;
+
+/// The parameter for the contract function `init` which sets up the contract.
+#[derive(Serial, Deserial, SchemaType)]
+struct InitParams {
+    /// pay royalties on this contract
+    pay_royalty:  bool,
+}
+/// The parameter for the contract function `mint` which mints a number of
+/// token types and/or amounts of tokens to a given address.
+#[derive(Serial, Deserial, SchemaType)]
+struct MintParams {
+    /// Owner of the newly minted tokens.
+    owner:  Address,
+    /// A collection of tokens to mint.
+    tokens: collections::BTreeMap<ContractTokenId, ContractTokenAmount>,
+}
+
+/// The parameter type for the contract function `setImplementors`.
+/// Takes a standard identifier and a list of contract addresses providing
+/// implementations of this standard.
+#[derive(Debug, Serialize, SchemaType)]
+struct SetImplementorsParams {
+    /// The identifier for the standard.
+    id:           StandardIdentifierOwned,
+    /// The addresses of the implementors of the standard.
+    implementors: Vec<ContractAddress>,
+}
+
+
+/// The parameter type for the contract function `check_royalty`.
+/// Takes a tokenID and a sale price.
+#[derive(Debug, Serialize, SchemaType)]
+struct CheckRoyaltyParams {
+    /// The identifier of the token.
+    id:      ContractTokenId,
+    /// The sale price.
+    sale_price: u64,
```

Comment:
In case of arithmetic functions (e.g.`+`, `-`,`/`, ...) work with `TokenAmountU64`, would it make it easier to use `TokenAmountU64` here?